SimCore/src/SimCore/LHE/LHEReader.cxx

```diff
@@ -1,18 +1,19 @@
 #include "SimCore/LHE/LHEReader.h"
 
-// STL
-#include <iostream>
-
-namespace simcore::lhe {
+namespace simcore {
+namespace lhe {
 
 LHEReader::LHEReader(std::string& filename) {
-  std::cout << "Opening LHE file " << filename << std::endl;
+  ldmx_log(info) << "Opening LHE file " << filename;
   ifs_.open(filename.c_str(), std::ifstream::in);
-}
 
-LHEReader::~LHEReader() { ifs_.close(); }
+  if (!ifs_.is_open()) {
+    ldmx_log(fatal) << "Failed to open LHE file: " << filename;
+    throw std::runtime_error("Failed to open LHE file: " + filename);
```

Comment:
We are in ldmx-sw, so we can throw a framework exception.

```suggestion
    EXCEPTION_RAISE("BadFile", "Failed to open LHE file: "+filename);
```

(context: I don't throw framework exceptions in G4DarkBreM because we wanted it to be able to be compiled separately from ldmx-sw.)